src/detect-distance.c

```diff
@@ -218,25 +194,22 @@ static int DetectDistanceTest01(void)
  */
 static int DetectDistanceTestPacket01 (void)
 {
-    int result = 0;
     uint8_t buf[] = { 0x01, 0x00, 0x00, 0x00, 0x00, 0x00 };
     uint16_t buflen = sizeof(buf);
     Packet *p;
     p = UTHBuildPacket((uint8_t *)buf, buflen, IPPROTO_TCP);
 
-    if (p == NULL)
-        goto end;
-
+    FAIL_IF_NULL(p);
     char sig[] = "alert tcp any any -> any any (msg:\"suricata test\"; "
                     "byte_jump:1,2; content:\"|00|\"; "
                     "within:1; distance:2; sid:98711212; rev:1;)";
 
     p->flowflags = FLOW_PKT_ESTABLISHED | FLOW_PKT_TOCLIENT;
-    result = UTHPacketMatchSig(p, sig);
+    UTHPacketMatchSig(p, sig);
```

Comment:
How can the check fail now?

Reply:
You could just directly call the FAIL macro with this call, say:
`FAIL_IF_NOT(UTHPacketMatchSig(p, sig));`

If you do `git grep UTHPacketMatchSig` on the repository, you'll find a few examples.